src/rates/testnetMarketMaker.ts

```diff
@@ -105,16 +135,55 @@ export default class TestnetMarketMaker implements TradeEvaluationService {
       return false;
     }
 
+    await this.checkSufficientFunds(sellAsset, sellNominalAmount);
+
     const currentBuyRate = buyBalance.div(sellBalance);
     const tradeBuyRate = buyNominalAmount.div(sellNominalAmount);
 
     return tradeBuyRate >= currentBuyRate;
   }
 
-  private async checkSufficientFunds(sellAsset: Asset) {
-    const balance = await this.balanceLookups[sellAsset]();
+  private async checkSufficientFunds(asset: Asset, nominalAmount?: Big) {
+    const balance = await this.balanceLookups[asset]();
+
     if (balance.eq(0)) {
-      throw new Error("Insufficient funds");
+      logger.error(`Insufficient funds, asset ${asset} balance is 0`);
+      return false;
+    }
+
+    if (nominalAmount) {
+      const balanceSufficientForAmount = balance.sub(nominalAmount);
+      if (balanceSufficientForAmount.lte(0)) {
+        logger.error(
+          `Insufficient funds, asset ${asset} balance is ${balance} but trade requires ${nominalAmount}`
+        );
+        return false;
+      }
+    }
+
+    this.lowBalanceWarning(balance, asset, new Big(0.2));
```

Comment:
🔧 Maybe add it as a configuration to the file? :)

Reply:
this is worked in